tests/test_rest.py

```diff
@@ -443,5 +445,168 @@ def test_yaml(self):
         self.assertEqual(res.body, expected)
 
 
+class TestSchemaDecorator(unittest.TestCase):
+
+    def setUp(self):
+        self.root_app = bottle.default_app()
+        self.app = webtest.TestApp(self.root_app)
+
+        def callback(body=None, query=None):
+            return dict(body=body, query=query)
+        self.callback = callback
+
+    def test_no_schema(self):
+        # Use the decorator without the @ syntax:
+        self.callback = rest.schema()(self.callback)
+
+        self.root_app.route('/foo', callback=self.callback)
```

Comment:
Does this route get overwritten in the same app for every test since setup uses the `default_app()`? Maybe use `self.root_app = bottle.Bottle()` to create a new instance for each tests is "safer"? I'm not seeing issues, but it _might_ break intermittently or hide problems if/when any tests run concurrently or some garbage is left over in the app...